cmd/export_ledger_entry_changes.go

```diff
@@ -18,6 +25,97 @@ confirmed by the Stellar network.
 If no data type flags are set, then by default all of them are exported. If any are set, it is assumed that the others should not
 be exported.`,
 	Run: func(cmd *cobra.Command, args []string) {
+		startNum, endNum, _, _, _ := utils.MustBasicFlags(cmd.Flags(), cmdLogger)
+		execPath, configPath, exportAccounts, exportOffers, exportTrustlines, _ := utils.MustCoreFlags(cmd.Flags(), cmdLogger)
+
+		//if none of the export flags are set, then we assume that everything should be exported
+		if !exportAccounts && !exportOffers && !exportTrustlines {
+			exportAccounts, exportOffers, exportTrustlines = true, true, true
+		}
+
+		if configPath == "" && endNum == 0 {
+			cmdLogger.Fatal("stellar-core needs a config file path when exporting ledgers continuously (endNum = 0)")
+		}
+
+		var err error
+		execPath, err = filepath.Abs(execPath)
+		if err != nil {
+			cmdLogger.Fatal("could not get absolute filepath for stellar-core executable: ", err)
+		}
+
+		configPath, err = filepath.Abs(configPath)
+		if err != nil {
+			cmdLogger.Fatal("could not get absolute filepath for the config file: ", err)
+		}
+
+		core, err := input.PrepareCaptiveCore(execPath, configPath, startNum, endNum)
+		if err != nil {
+			cmdLogger.Fatal("error creating a prepared captive core instance: ", err)
+		}
+
+		accChannel, offChannel, trustChannel := createChangeChannels(exportAccounts, exportOffers, exportTrustlines)
+		input.StreamChanges(core, startNum, endNum, accChannel, offChannel, trustChannel)
+
+		transformedAccounts := make([]transform.AccountOutput, 0)
+		transformedOffers := make([]transform.OfferOutput, 0)
+		transformedTrustlines := make([]transform.TrustlineOutput, 0)
+
+		for {
```

Comment:
I completely miswrote this earlier. Refactor this code so the blocking `for-select` is in a separate function (`input.ReceiveChanges`), that function should return the three slices and an error. This will be functionally the same but significantly clean up the command-line code.